api-docs/client-interactions.md

```diff
@@ -0,0 +1,198 @@
+      "interactionType": "by phone",
+      "notes": "Application successfully filled out. Decision pending.",
+      "startTime": "2019-05-06T05:00:00.000Z",
+      "endTime": "2019-05-06T06:00:00.000Z",
+      "location": null,
+      "referenceServices": ["Supermarket Visit"],
+      "canModify": true,
+      "isDeleted": false,
+      "createdBy": {
+        "userId": 1,
+        "firstName": "Joel",
+        "lastName": "Denning",
+        "fullName": "Joel Denning",
+        "timestamp": "2019-05-06T06:00:00.000Z"
+      }
+    },
+    {
+      "id": 2,
+      "title": "Financial Counseling - Financial Coach Referal",
+      "interactionType": "in person",
+      "notes": "Appointment setup with financial coach. Client would like to save up enough for a down payment on a car better suited for their work.",
+      "startTime": "2019-05-06T05:00:00.000Z",
+      "endTime": "2019-05-06T06:00:00.000Z",
+      "location": "CU Office",
+      "referenceServices": ["DACA", "Youth Group"],
+      "canModify": true,
+      "isDeleted": false,
+      "createdBy": {
+        "userId": 1,
+        "firstName": "Joel",
+        "lastName": "Denning",
+        "fullName": "Joel Denning",
+        "timestamp": "2019-05-06T06:00:00.000Z"
+      }
+    }
+  ]
+}
+```
+
+## Create a Client Interaction
+
+### Request
+
+```http
+POST /api/clients/:id/interactions
+```
+
+```json
+{
+  "title": "Financial Counseling - Financial Coach Referal",
+  "interactionType": "in person",
+  "notes": "Appointment setup with financial coach. Client would like to save up enough for a down payment on a car better suited for their work.",
+  "startTime": "2019-05-06T05:00:00.000Z",
+  "endTime": "2019-05-06T06:00:00.000Z",
+  "location": "CU Office",
+  "referenceServices": ["DACA", "Youth Group"],
+  "canModify": true,
+  "isDeleted": false,
+  "createdBy": {
+    "userId": 1,
+    "firstName": "Joel",
+    "lastName": "Denning",
+    "fullName": "Joel Denning",
+    "timestamp": "2019-05-06T06:00:00.000Z"
+  }
+}
+```
+
+**_Notes_**
+
+- `program` is an enum with the following possible string values `Preventive Health - VDS daily services`, `Preventive Health - VDS mobile`, `Preventive Health - VDS Sabath`, `Preventive Health - Mobile Health Services`, `Preventive Health - Other`, `Nutrition - Nutrition`, `Nutrition - Supermarket visit`, `Nutrition - Cooking class`, `Nutrition - SNAP`, `Immigration - Citizenship`, `Immigration - Family Petition`, `Immigration - DACA`, `Immigration - Other`, `Financial Counseling`, `Worker Rights`, `Youth Group`, `Community Organization - Volunteering`, `Community Organization - Promoter`, `Community Organization - Internship`, `Family Support Services - LDS Vouchers`, `Family Support Services - Faxes`, `Family Support Services - References outside of CU`, `Family Support Services - Other`
+
+- `interactionType` is an enum with the possible values of `in person`, `by phone`, `workshop/talk`, `one on one/light touch`, `consultation`
```

Comment:
I would recommend making the enum values be camel cased instead of having spaces. This is for consistency with other enums in our database. See https://github.com/JustUtahCoders/comunidades-unidas-internal/blob/a59cb7cd777dfa146ff4f95b41714fc4947dd2a2/migrations/sqls/20190629230418-activity-log-up.sql#L6 and https://en.wikipedia.org/wiki/Camel_case

Reply:
This still needs to be addressed